controller/src/main/java/io/pravega/controller/store/stream/tables/TableHelper.java

```diff
@@ -645,4 +645,39 @@ public static boolean isRerunOf(final List<Integer> segmentsToSeal,
 
         return historyRecordOpt;
     }
+
+    public static boolean isScaleInputValid(final List<Integer> segmentsToSeal,
+                                            final List<AbstractMap.SimpleEntry<Double, Double>> newRanges,
+                                            final byte[] segmentTable) {
+
+        List<AbstractMap.SimpleEntry<Double, Double>> oldRanges = segmentsToSeal.stream()
+                .map(segment -> SegmentRecord.readRecord(segmentTable, segment)
+                        .map(x -> new AbstractMap.SimpleEntry<>(x.getRoutingKeyStart(), x.getRoutingKeyEnd())))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toList());
+
+        return reduce(oldRanges).equals(reduce(newRanges));
+    }
+
+    private static List<AbstractMap.SimpleEntry<Double, Double>> reduce(List<AbstractMap.SimpleEntry<Double, Double>> ranges) {
+        List<AbstractMap.SimpleEntry<Double, Double>> result = new ArrayList<>();
+        double low = -1.0;
+        double high = -1.0;
+        for (AbstractMap.SimpleEntry<Double, Double> range : ranges) {
+            if (high < range.getKey()) {
+                if (low != -1.0 && high != -1.0) {
+                    result.add(new AbstractMap.SimpleEntry<>(low, high));
+                }
+                low = range.getKey();
+                high = range.getValue();
+            } else {
+                high = range.getValue();
```

Comment:
This method reduces a given range by merging adjacent ranges into a single range. 
For example: lets say we have 3 ranges {[0, 0.2), [0.2, 0.4), [0.6, 0.8)}
The reduced value of this set will be {[0, 0.4), [0.6, 0.8)}
Now any of the infinite possible "newRanges" that reduce to the 'reduced' set of segmentsToSeal is deemed a valid input. 

This reduce method computes the reduced set from an original set. We first sort the input set by key.low.
Then we compute adjacent iff prev.high = next.low. 
So {[0.2, 0.4) [0.3, 0.6)} will be reduced to {[0.2, 0.4) [0.3, 0.6)} 
and {[0.2, 0.4) [0.4, 0.6)} will be reduced to {[0.2, 0.6)}